### PR TITLE
feat: remove runtime dependency on api key by using token provider

### PR DESCRIPTION
**Pull request by rsioss**
added @OlafConijn as a review on this one -- I copy 🍝 your changes from one of the archived demo repos, so wanted to make sure you agree that it looks good

### CODE REVIEW

File: src/common/constants.ts [hunk 1]
```diff
@@ -1,8 +1,19 @@
 import { NodeHttpHandler } from "@aws-sdk/node-http-handler";
-import { requiredEnvVar } from "./environment.js";
+import { fromAwsCredentialIdentity } from "@stedi/sdk-token-provider-aws-identity";
 
-export const DEFAULT_SDK_CLIENT_PROPS = {
-  apiKey: requiredEnvVar("STEDI_API_KEY"),
+interface ClientConfig {
```
Comment (rsioss) [inline src/common/constants.ts:4]: added this interface to appease eslint (it didn't like the inferred type of `DEFAULT_SDK_CLIENT_PROPS` below after adding the `...credentials` destructuring)

File: src/deploy/deploy-resources.ts [hunk 2]
```diff
@@ -75,7 +75,6 @@ You can generate a new API key here: https://www.stedi.com/app/settings/api-keys
   const stediAccountId = await resolveAccountId();
   const functionPaths = getFunctionPaths(pathMatch);
 
-  const FUNCTION_NAMES: string[] = [];
```
Comment (rsioss) [inline src/deploy/deploy-resources.ts:78]: had an IDE warning for this... the value is set but never read

**Review by BDQ — APPROVED** [review]

**Review by OlafConijn — CHANGES_REQUESTED** [review]
please hold of mergning untill we have an updated version of the `@stedi/sdk-token-provider-aws-identity` lib

**Review by OlafConijn — APPROVED** [review]
approved. there found a solution around having to update the client SDKs. all good, apologies for the churn.

**Pull request merged**